src/upsample.jl

```diff
@@ -180,9 +180,9 @@ function upsample_bilinear_whcn!(output::AbstractArray{T,4}, input::AbstractArra
     out_w, out_h, _, _ = size(output)
     output_slice_size = out_h * out_w
 
-    # T() and // so that we can handle rationals (super slow)
-    width_scale  = T((in_w - 1) // (out_w - 1))
-    height_scale = T((in_h - 1) // (out_h - 1))
+    #real(T)() and // so that we can handle rationals (super slow)
+    width_scale  =real(T)((in_w - 1) // (out_w - 1))
```

Comment:
Do you mind extracting out `real(T)` as `RT` or some like name? Otherwise this LGTM.

Reply:
Done